pkg/ephemeral-disk-utils/utils.go

```diff
@@ -22,13 +22,11 @@ package ephemeraldiskutils
 import (
 	"fmt"
 	"os"
-	"os/user"
-	"strconv"
 	"syscall"
 )
 
 // TODO this should be part of structs, instead of a global
-var DefaultOwnershipManager OwnershipManagerInterface = &OwnershipManager{user: "qemu"}
+var DefaultOwnershipManager OwnershipManagerInterface = &OwnershipManager{uid: 107, gid: 107}
```

Comment:
I remember there was a discussion about `OwnershipManager` in one of the PRs (it was some time ago, cannot find it now). If I am not mistaken, @davidvossel proposed to use an init-container based on `virt-launcher` to propagate the UID/GID of the `qemu` user to `virt-handler` (similarly to how its now done with node-labeler). Maybe it would be better to go that way? IMHO having the IDs hardcoded is not a good solution either.

Reply:
I think it would make sense to have a cluster-wide knob for the uid/gid. The second option would be to use the launcher`s database to lookup the qemu user.  I guess it depends on the flexibility we would agree on.

Reply:
> If I am not mistaken, @davidvossel proposed to use an init-container based on `virt-launcher` to propagate the UID/GID of the `qemu` user to `virt-handler` (similarly to how its now done with node-labeler and libvirt capabilities). Maybe it would be better to go that way? IMHO having the IDs hardcoded is not a good solution either.

That sounds rather complicated. I think 107 for qemu is pretty much a de facto standard.

> IMHO having the IDs hardcoded is not a good solution either.

Yes that does not really help if it is not propagated from a single point of truth. I a gree.

Reply:
> I think it would make sense to have a cluster-wide knob for the uid/gid. The second option would be to use the launcher`s database to lookup the qemu user. I guess it depends on the flexibility we would agree on.

virt-handler could just look it up from `/proc/<launcher-pid>/root/etc/passwd`. But I am not sure if we can't just assume `qemu` to be `107` until we have a use-case which requires us to consistently change these values. But I can definitely see that it could lead to problems in corner cases.

Reply:
> > I think it would make sense to have a cluster-wide knob for the uid/gid. The second option would be to use the launcher`s database to lookup the qemu user. I guess it depends on the flexibility we would agree on.
> 
> virt-handler could just look it up from `/proc/<launcher-pid>/root/etc/passwd`.

Virt-laucnher will have to be running already to check this, but in some cases, we will need to change ownership before it is running.

> But I am not sure if we can't just assume `qemu` to be `107` until we have a use-case which requires us to consistently change these values. But I can definitely see that it could lead to problems in corner cases.

That's my take on this as well. At least at this point, I think it's better for us to just assume that `qemu` is `107`
I saw that there are several places in the code where we just use this as a constant.

Reply:
The problem with hardcoded uid may arise when building custom images for KubeVirt. We faced that issue some time ago:
https://github.com/kubevirt/kubevirt/commit/03e7df09bfc8b5107b382130d77bc57089a8adb0 (back then it was in virt-launcher though).

Probably most of the distros are now using uid `107` for `qemu` but AFAIK it is **not** sort of a standard. In general I agree that the lookup of `qemu` user in `virt-handler` is not guaranteed to work smoothly in all cases and it introduces an unneded dependency. Though switching back to the hardcoded uid does not seem like a long-term solution.

Reply:
Hm, we even suggest users to create container disks like this: https://kubevirt.io/user-guide/virtual_machines/disks_and_volumes/#containerdisk-workflow-example:

```
cat << END > Dockerfile
FROM scratch
ADD --chown=107:107 fedora25.qcow2 /disk/
END
```

Without this, we risk that overlay storage which copies the whole file on permission changes (some storage does not treat file metadata independent of the content), which makes containerDisks pointless then.

Maybe we could instead recommend 777 as permission, but we are at least already knee deep in assuming 107 :/

Reply:
Regarding `--chown=107:107`: I wonder why not setting the needed permissions in runtime instead? Cant `virt-handler` do that for containerdisks so the user does not have to set the ownership when creating an image?

Reply:
> Regarding `--chown=107:107`: I wonder why not setting the needed permissions in runtime instead? Cant `virt-handler` do that for containerdisks so the user does not have to set the ownership when creating an image?

If the overlay storage supports copying file metadata only when only metadata gets modified, then yes. Just for reference what I am talking about (https://www.kernel.org/doc/html/latest/filesystems/overlayfs.html#metadata-only-copy-up).

Reply:
ah, I see. I somehow missed the overlay part in the prev message. makes sense then.

Reply:
For now, my recommendation is that we assume 107 is the qemu user, but source that uid/gid from a single source of truth in virt-handler... so not having references to 107 spread out all over the place. If we refactor the code in this way, then if we ever do need to transition to a more complex method where the qemu uid/gid needs to be discovered from virt-launcher, it won't be a big deal.

so pragmatically, i don't think that hard coding 107 once in virt-handler as the expected qemu uid/gid is all that bad. If there's evidence in the future that this assumption doesn't hold true, we'll proceed with the more complex logic of introspecting the virt-launcher pod.

sound reasonable?